src/metal.cs

```diff
@@ -2050,5 +2399,243 @@ interface MTLStageInputOutputDescriptor : NSCopying
+		[Export ("device")]
+		IMTLDevice Device { get; }
+
+		[Abstract]
+		[NullAllowed, Export ("commandQueue")]
+		IMTLCommandQueue CommandQueue { get; }
+	}
+
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface MTLCaptureManager
+	{
+		[Static]
+		[Export ("sharedCaptureManager")]
+		MTLCaptureManager SharedCaptureManager { get; }
+
+		[Export ("newCaptureScopeWithDevice:")]
+		IMTLCaptureScope CreateNewCaptureScope (IMTLDevice device);
+
+		[Export ("newCaptureScopeWithCommandQueue:")]
+		IMTLCaptureScope CreateNewCaptureScope (IMTLCommandQueue commandQueue);
+
+		[Export ("startCaptureWithDevice:")]
+		void StartCapture (IMTLDevice device);
+
+		[Export ("startCaptureWithCommandQueue:")]
+		void StartCapture (IMTLCommandQueue commandQueue);
+
+		[Export ("startCaptureWithScope:")]
+		void StartCapture (IMTLCaptureScope captureScope);
+
+		[Export ("stopCapture")]
+		void StopCapture ();
+
+		[NullAllowed, Export ("defaultCaptureScope", ArgumentSemantic.Strong)]
+		IMTLCaptureScope DefaultCaptureScope { get; set; }
+
+		[Export ("isCapturing")]
+		bool IsCapturing { get; }
+	}
+
+	[Mac (10, 13, onlyOn64: true), NoiOS, NoTV, NoWatch]
+	enum MTLDeviceNotificationName {
+		[Field ("MTLDeviceWasAddedNotification")]
+		DeviceWasAdded,
+
+		[Field ("MTLDeviceRemovalRequestedNotification")]
+		DeviceRemovalRequested,
+
+		[Field ("MTLDeviceWasRemovedNotification")]
+		MTLDeviceWasRemoved
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	interface MTLPipelineBufferDescriptor : NSCopying
+	{
+		[Export ("mutability", ArgumentSemantic.Assign)]
+		MTLMutability Mutability { get; set; }
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	interface MTLPipelineBufferDescriptorArray
+	{
+		[Export ("objectAtIndexedSubscript:")]
+		MTLPipelineBufferDescriptor GetObject (nuint bufferIndex);
+
+		[Export ("setObject:atIndexedSubscript:")]
+		void SetObject ([NullAllowed] MTLPipelineBufferDescriptor buffer, nuint bufferIndex);
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	interface MTLArgumentDescriptor : NSCopying
+	{
+		[Static]
+		[Export ("argumentDescriptor")]
+		MTLArgumentDescriptor ArgumentDescriptor { get; }
+
+		[Export ("dataType", ArgumentSemantic.Assign)]
+		MTLDataType DataType { get; set; }
+
+		[Export ("index")]
+		nuint Index { get; set; }
+
+		[Export ("arrayLength")]
+		nuint ArrayLength { get; set; }
+
+		[Export ("access", ArgumentSemantic.Assign)]
+		MTLArgumentAccess Access { get; set; }
+
+		[Export ("textureType", ArgumentSemantic.Assign)]
+		MTLTextureType TextureType { get; set; }
+
+		[Export ("constantBlockAlignment")]
+		nuint ConstantBlockAlignment { get; set; }
+	}
+
+	interface IMTLArgumentEncoder { }
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[Protocol, Model]
```

Comment:
From the documentation it looks like only Apple can create instances of this protocol, which means `[Model]` (and `[BaseType]`) are not needed.